authentication/app.js

```diff
@@ -0,0 +1,259 @@
+// See README.md for details about this app
+
+const fs = require('fs').promises
+const http = require("http")
+const https = require("https")
+var url = require('url')
+
+const CLIENT_ID = process.env.CLIENT_ID
+const CLIENT_SECRET = process.env.CLIENT_SECRET
+const CLIENT_SCOPE = process.env.SCOPE || 'read_fields'
+const CLIENT_STATE = Math.random().toString(36).substr(2) // a random token generated by the client app, protects against CSRF
+
+const APP_PROTOCOL = 'http'
+const APP_HOST = 'localhost'
+const APP_PORT = 8000
+const APP_BASE_URL = `${APP_PROTOCOL}://${APP_HOST}:${APP_PORT}`
+const APP_CALLBACK_URI = `${APP_BASE_URL}/authenticate/callback`
+
+const FIELDAGENT_PROTOCOL = process.env.FIELDAGENT_PROTOCOL || 'https'
+const FIELDAGENT_HOST = process.env.FIELDAGENT_HOST || 'api.sentera.com'
+const FIELDAGENT_PORT = process.env.FIELDAGENT_PORT || ''
+const FIELDAGENT_BASE_URL = `${FIELDAGENT_PROTOCOL}://${FIELDAGENT_HOST}:${FIELDAGENT_PORT}`
```

Comment:
Does the `:` in `:${FIELDAGENT_PORT}` need to be conditional based on non-empty `FIELDAGENT_PORT`?

Reply:
I think you are right. I meant to make that change during my development and local testing, and forgot to do that. Thanks for pointing this out.

BTW - I've only tested this app with my local CloudVault server.  There's a change in https://github.com/SenteraLLC/CloudVault/pull/3159 that needs to get deployed to staging first before I can run this sample app against a non-local CloudVault server.

Reply:
Done! https://github.com/SenteraLLC/code-examples/pull/1/commits/f530079a378aaae23fc5e9cfadc40d421b08d72e